syntax = "proto3";

package flom.proto;

import "frame.proto";

message EffectorType {
  enum Type {
    World = 0;
    Local = 1;
  }
  Type location = 1;
  Type rotation = 2;
}

message Motion {
  string model_id = 1;
  enum Loop {
    None = 0;
    Wrap = 1;
  }
  Loop loop = 2;
  map<string, EffectorType> effector_types = 3;
  repeated Frame frames = 4;
}
